src/BattleServer/abilities.cpp

```diff
@@ -3021,16 +3021,12 @@ struct AMBattleBond : public AM {
     }
 
     static void ak(int s, int, BS &b) {
-        if (b.koed(s))
+        if (b.koed(s) || b.poke(s).num() != Pokemon::Greninja)
```

Comment:
Might need a check for && b.poke(s) != Pokemon::Ash_Greninja

I think otherwise it will send a message about transformation everytime ash greninja gets a ko.